src/openloco/ui/WindowManager.cpp

```diff
@@ -273,7 +269,63 @@ namespace openloco::ui::WindowManager
     // 0x004C6118
     void update()
     {
-        call(0x004C6118);
+        _tooltipNotShownTicks = _tooltipNotShownTicks + time_since_last_tick;
+
+        // Has to do with graphics mode switching. Remove?
+        _9DA3D4 = _9DA3D4 + 1;
+        if (_9DA3D4 == 224 && ui::dirty_blocks_initialised())
+        {
+            if (*_callingWindowType != WindowType::undefined)
+            {
+                _9DA3D4 = _9DA3D4 - 1;
+            }
+            else
+            {
+                // set_window_pos_wrapper();
+            }
+        }
+
+        if (!ui::dirty_blocks_initialised())
+        {
+            return;
+        }
+
+        if (!intro::is_active())
+        {
+            gfx::draw_dirty_blocks();
+        }
+
+        for (ui::window* w = &_windows[0]; w != _windowsEnd; w++)
+        {
+            w->viewports_update_position();
+        }
+
+        // 1000 tick update
+        _thousandthTickCounter = _thousandthTickCounter + time_since_last_tick;
+        if (_thousandthTickCounter >= 1000)
+        {
+            _thousandthTickCounter = 0;
+            for (ui::window* w = _windowsEnd - 1; w >= _windows; w--)
+            {
+                w->call_6();
```

Comment:
I believe we recently named this event `periodic update` over at https://github.com/OpenRCT2/OpenRCT2/pull/8684